components/org.wso2.carbon.extension.identity.sso.jwt/src/main/java/org/wso2/carbon/identity/sso/jwt/factory/JWTInboundIdentityResponseFactory.java

```diff
@@ -0,0 +1,164 @@
+     */
+    @Override
+    public HttpIdentityResponse.HttpIdentityResponseBuilder create(IdentityResponse identityResponse) {
+
+        HttpIdentityResponse.HttpIdentityResponseBuilder builder
+                = new HttpIdentityResponse.HttpIdentityResponseBuilder();
+
+        if (identityResponse instanceof JWTInboundResponse) {
+            JWTInboundResponse jwtInboundResponse = (JWTInboundResponse) identityResponse;
+
+            String logoutUrl = jwtInboundResponse.getLogoutUrl();
+            if (StringUtils.isBlank(jwtInboundResponse.getToken()) && StringUtils.isNotBlank(logoutUrl)) {
+                // Successful logout scenario.
+                if (log.isDebugEnabled()) {
+                    log.debug("Logout URL: " + JWTInboundUtil.neutralize(logoutUrl) + " provided and the token is set" +
+                            " to empty. Hence considering as successful logout scenario.");
+                }
+                builder.setStatusCode(HttpServletResponse.SC_FOUND);
+                // Redirect to Logout URL.
+                builder.setRedirectURL(logoutUrl);
+            } else if (StringUtils.isBlank(jwtInboundResponse.getToken())) {
+                // un-authenticated scenario without logout URL - Redirect to Retry.do - Show Client Error
+                String clientErrorPage = jwtInboundResponse.getEndpointUrl();
+                if (StringUtils.isNotBlank(clientErrorPage)) {
+                    builder.setStatusCode(HttpServletResponse.SC_FOUND);
+                    builder.setRedirectURL(clientErrorPage);
+
+                    Map<String, String[]> parameters = new HashMap<>();
+                    if (jwtInboundResponse.getParameters() != null) {
+                        for (Map.Entry<String, String> entry : jwtInboundResponse.getParameters().entrySet()) {
+                            parameters.put(entry.getKey(), new String[]{entry.getValue()});
+                        }
+                        builder.setParameters(parameters);
+                    }
+                } else {
+                    builder.setStatusCode(HttpServletResponse.SC_UNAUTHORIZED);
+                }
+            } else {
+                // Successful authenticated scenario.
+                if (log.isDebugEnabled()) {
+                    log.debug("Successful authenticated scenario. Building the response to redirect.");
+                }
+                builder.setStatusCode(HttpServletResponse.SC_FOUND);
+                String jwtParamName = jwtInboundResponse.getJwtParamName();
+                String jwtToken = jwtInboundResponse.getToken();
+                if (log.isDebugEnabled() &&
+                        IdentityUtil.isTokenLoggable(JWTInboundConstants.IdentityTokens.JWT_TOKEN)) {
+                    log.debug(
+                            "Adding JWT parameter: '" + JWTInboundUtil.neutralize(jwtParamName) + "' with the value: " +
+                                    JWTInboundUtil.neutralize(jwtToken));
```

Comment:
@omindu is it okay to log the token?

Reply:
We usually log the access token with isTokenLoggable enabled as defined at https://is.docs.wso2.com/en/5.10.0/setup/adding-logs-for-tokens/.